[7] chore: clean up TODOs after refactoring

shoom3301 opened the pull request:
# Summary

Figma: https://www.figma.com/file/66BWGSBbkqLYCig8gLKj29/Fee-0---Modal?type=design&node-id=290-275&mode=design&t=C0K7pAWDEUFqGayJ-0

PRs list: https://github.com/cowprotocol/cowswap/labels/Trade%20confirm

In this PR I will add fixes after testing and code review. This is the final PR in the refactoring.

  # To Test

Actually the whole app regression test.
The trade flow UI is changed, now all modals must be displayed as screens + UI changes from Figma.
TODO: fix e2e tests

anxolin reviewed (commented):
The UI looks way better!!
I really like the new UX.


@fairlighteth, could we try to add an UI effect when the counter arrives to zero? Like a simmering or lightning effect? It goes from 1 to 10 all the sudden. 

Alternatively, or in addition, we could use a donut like:

https://github.com/cowprotocol/cowswap/assets/2352112/acbaff08-3586-4ae0-a715-f7aa7a169e77



See
https://github.com/cowprotocol/cowswap/assets/2352112/c9e00a85-8e64-4589-8681-eaf053d4c9fd



@shoom3301 I think we need to close the confirmation modal if we change the network, look at what happens:
<img width="1544" alt="Screenshot at Feb 01 11-57-04" src="https://github.com/cowprotocol/cowswap/assets/2352112/a43c77a9-b179-49c4-8d56-c3d589e96640">

alfetopito commented:
I thought we were getting rid of this modal:
![image](https://github.com/cowprotocol/cowswap/assets/43217/b99cfa06-d11e-4740-b816-f4444d766bb6)

Not yet?

alfetopito commented:
There's a quote refresh on limit orders 
![image](https://github.com/cowprotocol/cowswap/assets/43217/eb69bb3f-d564-4cd9-b768-29d9b6c5d616)

I believe we agreed to not have it there?

alfetopito commented:
I have unlimited approval for COW, but it shows this number instead
![image](https://github.com/cowprotocol/cowswap/assets/43217/99e88f8b-9a20-4ee8-863d-8884cbe97717)

Approval tx https://sepolia.etherscan.io/tx/0xa331c756f98119603dbfe6235091836f0798c60f7ec37de7e1a14fa0b61bc746

alfetopito reviewed (commented):
Overall it's working great!

Left a few comments on the behavior, but nothing major AFAICT

shoom3301 commented:
>think we need to close the confirmation modal if we change the network, look at what happens:

@anxolin thanks! Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/f19b740194880202ff210a5d919364665d4fb711

shoom3301 commented:
>I thought we were getting rid of this modal:

@alfetopito sorry, I'm not aware about this. Could you give more info please? When and why we decided to remove the modal?

shoom3301 commented:
>There's a quote refresh on limit orders

@alfetopito thanks! Removed it: https://github.com/cowprotocol/cowswap/pull/3756/commits/82ab9bda3bea0497c597b86715195bbab9ae953b

shoom3301 commented:
>I have unlimited approval for COW, but it shows this number instead

@alfetopito good catch! Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/e8227403e89abc4bc72442bbe5907b373c8f034c

elena-zh commented:
Hey @shoom3301 , great job!
Some my notes: 
1. There is no 'error' modal when: call a sign transaction (the event is sent to the wallet), then press on the back arrow in the modal, then open MM and reject signing an order
![no _ missalignment](https://github.com/cowprotocol/cowswap/assets/70885163/361fc989-8245-4ed4-a6b6-2a9847cdfe52)
2. Error modal header looks like missaligned a bit
3.  There is no 'error' modal when reject a Limit order signing (with/without pressing on  he back button on the confirm modal)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/d7917e71-82ae-45ce-832f-90306a234564)
4. No error message when reject ETH-flow order signing in Expert mode and in case I described in p.1
5. I see ETH token's approved qty in this PR
![image](https://github.com/cowprotocol/cowswap/assets/70885163/2bc53866-85aa-4168-bdca-86ad80bd6b07)
6. No error message when reject a token approval TX
7. For the Convert TX, should we show the back arrow on the waiting modal? 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/68551f61-b084-4c22-b897-78bc1be59d04)
8. Same question for tokens approvals from the Tokens page and Cancel order modals
![image](https://github.com/cowprotocol/cowswap/assets/70885163/01c1e641-cd3f-4e58-8012-eb1813152d15)
9. When a wallet is not connected, Approve button is in the loading state. It should be displayed and should call Connect to the wallet modal
![image](https://github.com/cowprotocol/cowswap/assets/70885163/f163a21a-39cb-4c5c-88e9-820aae39b6ce)
10. I don't see different types of warnings on the confirm modal: fees amount
![image](https://github.com/cowprotocol/cowswap/assets/70885163/56950e5f-e8b3-46ad-933a-279185628851)
11. Custom recipient warning (swap) + there is no recipient field on the modal: 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/1f0342f7-f1af-44b7-87a1-ad2e7ea0caa4)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/82f85779-4c1c-4c4c-9f19-143c3691945c)
12. Not possible to calculate price impact - swap
13. When I start signing the order, and close 'price impact warning', I get this 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/a66ac9c0-8d38-4a4d-9d32-37577b0743c9)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/9bbc3d58-853f-4c9a-be21-125af004410a)
14. Mobile view: modals do not look good in iOS
![image](https://github.com/cowprotocol/cowswap/assets/70885163/079c490c-4c0d-4b34-b928-62f17eb668d5)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/8b5c1bc5-1afc-4ecc-8d59-ddc54f5d565b)
15. It would be great to scroll to the top in a mobile view when placing a limit order: the Review button is at the bottom, the confirm modal appears at the top
16. ETH-flow process: looks like circles are missaligned a bit
![image](https://github.com/cowprotocol/cowswap/assets/70885163/e5dd0626-cfc5-401d-8aa2-cdede4b573b1)

17. No 'pending orders' pop-up when close the progress bar modal 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/a994ec71-1d82-40e2-aed7-b71181b31e0b)
18. (Leandro reported this as well): fully approved token shows an amount 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/58ce2edb-baf3-4a9b-9e08-a850a54e5672)
19. I've got a crash when I switched from Tokens page in Sepolia to the Mainnet
![image](https://github.com/cowprotocol/cowswap/assets/70885163/23334ac4-1e9e-4205-8a6d-da607209a441)
20. No error when I reject wrapping for the classic eth-flow 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/3aa94bab-956d-4f33-971a-cc89516dfd1b)
21. no waiting for confirm modal when place an order in Expert mode (we have it on Prod)
22. Approve action for the classic ETH-flow is broken 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/98ad79ef-397f-4b7a-a07a-37a9af196932)
23. Safe+bundling: the button's name on the confirm modal should be 'Confirm (Approve WETH and Swap)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/7599da77-484e-488e-98b4-d4df9a881814)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/17b47feb-0d5a-4ac1-a825-bed4e815d977)

24. Is this OK to see 'Unwrapping' modal outside the widget in Safe? 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/53a474f2-b180-40b7-884e-44ca264e29da)

Thanks!

alfetopito commented on apps/cowswap-frontend/src/modules/trade/containers/TradeConfirmModal/index.tsx:
```diff
@@ -93,7 +91,7 @@ function InnerComponent(props: InnerComponentProps) {
 
   if (error) {
-    return <TransactionErrorContent mode="screen" message={error} onDismiss={onDismiss} />
+    return <TransactionErrorContent isScreenMode={true} message={error} onDismiss={onDismiss} />
```
Nitpick:

```suggestion
    return <TransactionErrorContent isScreenMode message={error} onDismiss={onDismiss} />
```

alfetopito commented:
> > I thought we were getting rid of this modal:
> 
> @alfetopito sorry, I'm not aware about this. Could you give more info please? When and why we decided to remove the modal?

I understood the point of the new confirmation styles was to keep them visible while the user signed the order, so they could check the amounts being signed.
Thus, this intermediary sign modal would be removed.

shoom3301 commented:
@elena-zh thanks a lot for the comments!

1. Not a new issue, works the same on prod
2. Fixed
3. Not a new issue, works the same on prod
4. Not a new issue, works the same on prod
5. Fixed
6. Fixed
7. Fixed
8. Fixed
9. Fixed
10. Fixed
11. Fixed
12. Fixed
13. Fixed (p.s. it's reproducible on prod as well)
14. Fixed
15. Fixed
16. @fairlighteth could you help with it please?
17. Work in progress
18. Fixed
19. Fixed
20. Not a new issue, works the same on prod
21. Work in progress
22. Couldn't reproduce
23. Fixed
24. Couldn't reproduce

shoom3301 commented:
<img width="573" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/4235b14e-33a2-4e5a-96aa-3a89b2a79fe5">

<img width="564" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/1b8ffe07-1079-4aad-944f-34d358d7ef46">

<img width="561" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/a0f2ab13-e753-4796-bd73-8c9f43abe7f7">

<img width="558" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/02275e48-ae86-4921-b34b-5e30bb68147a">

<img width="554" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/657ffc25-d141-4d37-a05f-8e861f02a1ed">

@fairlighteth @elena-zh I also refactored classic eth-flow

shoom3301 commented:
@elena-zh could you please recheck zero-approve flow? (USDT)

alfetopito commented on apps/cowswap-frontend/src/common/pure/NewModal/index.tsx:
```diff
@@ -145,16 +151,22 @@ export interface NewModalProps {
   title?: string
   onDismiss?: () => void
   children?: React.ReactNode
+  screenMode?: boolean
```
Use `modalMode` to match the prop from the other stuff?

shoom3301 replied:
Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/08afda61eca55c270ff0dac10aa8a68c5a88ef6d

alfetopito commented on apps/cowswap-frontend/src/common/pure/TransactionErrorContent/index.tsx:
```diff
@@ -40,21 +41,23 @@ const AlertIcon = styled(AlertTriangle)`
   message: JSX.Element | string
   onDismiss(): void
-  mode?: 'screen' | 'modal'
+  isScreenMode?: boolean
```
Same here. I'd prefer to have a standard name for this.

shoom3301 replied:
Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/08afda61eca55c270ff0dac10aa8a68c5a88ef6d

elena-zh commented:
Hey @shoom3301 , great job, thank you!
Some issues that I still face: 
1. Error signing order is not detected on Limit orders page (placing modal hangs)
![limit order reject](https://github.com/cowprotocol/cowswap/assets/70885163/9e6f00fc-760f-4fd7-9aae-bfc404479175)
2. I see 'back' arrow on the token approval modal on the Tokens page
![image](https://github.com/cowprotocol/cowswap/assets/70885163/2d7222e3-6d19-42ac-8778-875df6ce6ccf)
3. When a wallet is not connected, you hidden the Approve button. But it exists on Prod and calls the Connect to wallet modal :)
![approve](https://github.com/cowprotocol/cowswap/assets/70885163/07fbb622-9555-46fe-a035-258f5eaa6632)
4. Custom recipient on Swap form: there is still no warning. See the example on the limit orders: 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/00920bd3-c4af-4fc9-8061-b098a0d98bd7)
5. Custom recipient warning on Limit orders: there is no need in 'Output will be send' message
![image](https://github.com/cowprotocol/cowswap/assets/70885163/2a6caeca-1e51-4203-8cf1-21301105a829)
Overall, it would be great to add a Recipient field to the modal on the Swap page and remove this message from there as well :)
6. I have reported #3788 issue for @fairlighteth to fix 16 point
7. approved token shows an amount of a token issue is not fixed

![image](https://github.com/cowprotocol/cowswap/assets/70885163/0f6821ba-b00f-47d9-92de-dcb292190dad)
8. Limit orders: call cancel order modal, press on the Back arrow. Reject signing cancellation request in MM. Call cancel order modal once again --> you will get 'Error signing order' modal
![cancel](https://github.com/cowprotocol/cowswap/assets/70885163/fb872d99-25e3-4320-acae-9141b95d432f)

Thanks!

alfetopito commented on apps/cowswap-frontend/src/modules/swap/containers/ConfirmSwapModalSetup/index.tsx:
```diff
@@ -57,9 +67,14 @@ export function ConfirmSwapModalSetup(props: ConfirmSwapModalSetupProps) {
+        <>
+          <TradeRates {...tradeRatesProps} isReviewSwap={true} />
+          <HighFeeWarning trade={tradeRatesProps.trade} />
+          {!priceImpact.priceImpact && <NoImpactWarning isAccepted={true} withoutAccepting={true} />}
```
Nitpick

```suggestion
          {!priceImpact.priceImpact && <NoImpactWarning isAccepted withoutAccepting />}
```

shoom3301 replied:
Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/08afda61eca55c270ff0dac10aa8a68c5a88ef6d

alfetopito commented on apps/cowswap-frontend/src/modules/usdAmount/hooks/useUsdPrice.ts:
```diff
@@ -30,5 +30,9 @@ export function useUsdPrice(currency: Nullish<Token>): UsdPriceState | null {
-  return usdPrices[currencyAddress] || null
+  const price = usdPrices[currencyAddress]
+
+  if (price && price.currency.chainId !== currency?.chainId) return null
+
+  return price || null
```
How about:

```suggestion
  if (!price || price.currency.chainId !== currency?.chainId) return null

  return price
```

alfetopito replied:
If we want to be extra compact:

```suggestion
  return !price || price.currency.chainId !== currency?.chainId ? null : price
```

But that might be less readable...

shoom3301 replied:
Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/08afda61eca55c270ff0dac10aa8a68c5a88ef6d

alfetopito reviewed (commented):
A few nitpicks in the code, nothing major.

elena-zh commented:
Hey @shoom3301 , one more weird case I've faced: 
1. Start placing an order with a permittable token (not approved)
2. When there is a request to sign a TX (approve) in the wallet, press back on the Approve modal
![image](https://github.com/cowprotocol/cowswap/assets/70885163/7c33c2e8-b903-449b-ad5f-dd05bb588972)

Placing order modal will appear (regular one). And it will not be possible to close it by pressing on the back button once again
![back](https://github.com/cowprotocol/cowswap/assets/70885163/a149b339-1837-4287-810d-c11aa3ecb1aa)


Same happens when press on the Back arrow from the 2nd screen (placing order after approval signing). And when I reject an order signing in the wallet. 

-----------------

Also, I have noticed that on the Swap form it says limit (for a permittable tokens flow): 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/31e7a5ca-469c-4987-bc06-65d5c3145f91)

elena-zh commented:
One more proposal: when place a swap order, there is a button at the bottom that navigates to the Game page. Maybe we should add an additional one ('Continue', like on the Limit orders) that will close the modal? I know that there is a 'back' arrow that closes it, but it is easier to press on the big button instead of the arrow :)
![continue](https://github.com/cowprotocol/cowswap/assets/70885163/35857df3-e507-46a4-8fa9-b1f4289cd98b)

@fairlighteth , WDYT?

shoom3301 commented:
@elena-zh thank you!
Fixed both issues

shoom3301 commented:
@elena-zh also fixed issues from https://github.com/cowprotocol/cowswap/pull/3756#issuecomment-1932141052,
besides https://github.com/cowprotocol/cowswap/issues/3788

shoom3301 commented on apps/cowswap-frontend/src/common/pure/NewModal/index.tsx:
```diff
@@ -145,16 +151,22 @@ export interface NewModalProps {
+        {!modalMode && (
+          <IconX onClick={onDismissCallback}>
+            <SVG src={CLOSE_ICON} />
+          </IconX>
```
Nice catch! Thanks!
Fixed

elena-zh reviewed (approved):
Looks good now!
I have created #3833  issue with some enhancements related to the Recipient value on the modal, but it can be addressed later.
